### PR TITLE
feat(rn): environment switcher

### PR DESCRIPTION
**Pull request by oliverlaz**
### Overview

More reliable environment switcher + webrtc 125.0.0-rc.2

### CODE REVIEW

File: sample-apps/react-native/dogfood/src/components/VideoWrapper.tsx
Comment (santhoshvai) [inline sample-apps/react-native/dogfood/src/components/VideoWrapper.tsx:1]: can we remove this please.. we dont all logs by default

File: sample-apps/react-native/dogfood/src/components/ParticipantsInfoList.tsx [hunk 1]
```diff
@@ -65,12 +66,24 @@ export const ParticipantsInfoList = ({
...
+          ? `https://pronto.getstream.io/join/${call.id}`
+          : environment === 'pronto-staging'
+            ? `https://pronto-staging.getstream.io/join/${call.id}`
+            : `https://getstream.io/video/demos/join/${call.id}`;
       await Share.share({
-        url: `https://stream-calls-dogfood.vercel.app/join/${call?.id}`,
+        url,
         title: 'Stream Calls | Join Call',
-        message: `Join me on the call using this link https://stream-calls-dogfood.vercel.app/join/${call?.id}`,
+        message: `Join me on the call using this link ${url}`,
```
Comment (santhoshvai) [inline sample-apps/react-native/dogfood/src/components/ParticipantsInfoList.tsx:86]: For deeplinks, more additionals are needed unforunately, for example in AndroidManifest

Comment (santhoshvai) [reply sample-apps/react-native/dogfood/src/components/ParticipantsInfoList.tsx:86]: but this is a start. Sharing alteast works

File: sample-apps/react-native/dogfood/src/components/VideoWrapper.tsx
Comment (santhoshvai) [reply sample-apps/react-native/dogfood/src/components/VideoWrapper.tsx:1]: Switcher already works, only the mode doesnt

File: sample-apps/react-native/dogfood/src/screens/LoginScreen.tsx [hunk 2]
```diff
@@ -123,14 +113,13 @@ const LoginScreen = () => {
           <View style={styles.header}>
             <Text style={styles.envText}>{t('Pronto')}</Text>
             <Switch
-              value={prontoEnvironment}
+              value={appEnvironment === 'pronto'}
```
Comment (santhoshvai) [inline sample-apps/react-native/dogfood/src/screens/LoginScreen.tsx:116]:
```suggestion
              value={appEnvironment === 'pronto' ||
              appEnvironment === 'pronto-staging'}
```

**Review by santhoshvai — APPROVED** [review]

File: sample-apps/react-native/dogfood/src/components/ParticipantsInfoList.tsx (hunk 1 above)
Comment (oliverlaz) [reply sample-apps/react-native/dogfood/src/components/ParticipantsInfoList.tsx:86]: that's another story... we should fix that one as well. Now, this makes it easy to join on Web when started in RN.

File: sample-apps/react-native/dogfood/src/screens/Meeting/MeetingScreen.tsx [hunk 3]
```diff
@@ -23,6 +23,8 @@ export const MeetingScreen = (props: Props) => {
     }
     return client.call(callType, callId);
   }, [callId, callType, client]);
+  // @ts-ignore
+  globalThis.call = call;
```
[the remark this answers is not in the document]
Comment (oliverlaz) [reply sample-apps/react-native/dogfood/src/screens/Meeting/MeetingScreen.tsx:27]: yes, this allows us to attach the new tools and access the current `call` instance straight from the console.

**Pull request merged**